tasmota/xsns_63_aht1x.ino

```diff
@@ -22,137 +22,160 @@
-  Wire.endTransmission();
-
-  delay(500);  // ?!?! too long
-
-  return (0x08 == (AHT10ReadStatus() & 0x68));
+  Wire.begin(aht1x_address);
+  Wire.beginTransmission(aht1x_address);
+  Wire.write(CalibrateCmd, 3);
+  if (Wire.endTransmission() != 0) return false;
+  delay(AHT1X_CMD_DELAY);
+
+  if((AHT1XReadStatus(aht1x_address) & 0x68) == 0x08)
+    return true;
+  return false;
 }
 
-uint8_t AHT10ReadStatus(void)
+unsigned char AHT1XReadStatus(uint8_t aht1x_address)
 {
-  Wire.requestFrom(AHT10_ADDR, 1);
-  uint8_t result = Wire.read();
+  unsigned char result = 0;
+  Wire.requestFrom(aht1x_address, (uint8_t) 1);
+  result = Wire.read();
   return result;
 }
 
-void AHT10Reset(void)
+void AHT1XReset(uint8_t aht1x_address)
 {
-  Wire.beginTransmission(AHT10_ADDR);
-  Wire.write(eSensorResetCmd);
+  Wire.beginTransmission(aht1x_address);
+  Wire.write(ResetCmd);
   Wire.endTransmission();
-  delay(20);
+  delay(AHT1X_RST_DELAY);
 }
 
 /********************************************************************************************/
-
-void AHT10Detect(void)
-{
-  if (I2cActive(AHT10_ADDR)) { return; }
-
-  if (AHT10Init()) {
-    I2cSetActiveFound(AHT10_ADDR, AHT10.name);
-    AHT10.count = 1;
-  }
-}
-
-void AHT10EverySecond(void)
+void AHT1XDetect(void)
 {
-  if (uptime &1) {
-    // AHT10: 55mS
-    if (!AHT10Read()) {
-      AddLogMissed(AHT10.name, AHT10.valid);
+  for (uint8_t i = 0; i < AHT1X_MAX_SENSORS; i++) {
+    if (I2cActive(aht1x_addresses[i])) { continue; }
+
+    if (AHT1XInit(aht1x_addresses[i]))
+    {
+      aht1x_sensors[aht1x_count].address = aht1x_addresses[i];
+      GetTextIndexed(aht1x_sensors[aht1x_count].types, sizeof(aht1x_sensors[aht1x_count].types), i, ahtTypes);
+      I2cSetActiveFound(aht1x_sensors[aht1x_count].address, aht1x_sensors[aht1x_count].types);
+      aht1x_count++;
     }
   }
 }
 
-void AHT10Show(bool json)
+void AHT1XShow(bool json)
 {
-  if (AHT10.valid) {
-    char temperature[33];
-    dtostrfd(AHT10.temperature, Settings.flag2.temperature_resolution, temperature);
-    char humidity[33];
-    dtostrfd(AHT10.humidity, Settings.flag2.humidity_resolution, humidity);
-
-    if (json) {
-      ResponseAppend_P(JSON_SNS_TEMPHUM, AHT10.name, temperature, humidity);
-#ifdef USE_DOMOTICZ
-      if ((0 == tele_period)) {
-        DomoticzTempHumSensor(temperature, humidity);
-      }
-#endif // USE_DOMOTICZ
-#ifdef USE_KNX
-      if (0 == tele_period) {
-        KnxSensor(KNX_TEMPERATURE, AHT10.temperature);
-        KnxSensor(KNX_HUMIDITY, AHT10.humidity);
-      }
-#endif // USE_KNX
-#ifdef USE_WEBSERVER
-    } else {
-      WSContentSend_PD(HTTP_SNS_TEMP, AHT10.name, temperature, TempUnit());
-      WSContentSend_PD(HTTP_SNS_HUM, AHT10.name, humidity);
-#endif // USE_WEBSERVER
+  for (uint8_t i = 0; i < aht1x_count; i++)
+  {
+     float tem,hum;
+     if (AHT1XRead(aht1x_sensors[i].address, tem, hum))
+     {
+       float aht_temp = ConvertTemp(tem);
+       float aht_hum = ConvertPressure(hum);
+
```

Comment:
Almost there but ConvertPressure should be ConvertHumidity.

Why don't you do it during AHT1XRead as I did. That way global info is updated every 2 seconds independant of when data is presented.

Reply:
ok, i rechange code complete to read the sensor every Second and only show the values if there are needed.
Thanks!